var/spack/repos/builtin/packages/saga-gis/package.py

```diff
@@ -38,31 +39,52 @@ class SagaGis(AutotoolsPackage):
 
     variant('gui',      default=True,   description='Build GUI and interactive SAGA tools')
     variant('odbc',     default=True,   description='Build with ODBC support')
+
     # FIXME Saga-gis configure file disables triangle even if
     # --enable-triangle flag is used
     # variant('triangle', default=True,   description='Build with triangle.c
+
     # non free for commercial use otherwise use qhull')
-    variant('libfire',  default=True,   description='Build with libfire (non free for commercial usage)')
-    variant('openmp',   default=True,   description='Build with OpenMP enabled')
-    variant('python',   default=False,  description='Build Python extension')
+    variant('libfire',      default=True,   description='Build with libfire (non free for commercial usage)')
+    variant('openmp',       default=True,   description='Build with OpenMP enabled')
+    variant('python',       default=False,  description='Build Python extension')
+
+    variant('postgresql',   default=False,  description='Build with PostgreSQL library')
+    variant('opencv',       default=False,  description='Build with libraries using OpenCV')
 
     depends_on('autoconf', type='build')
     depends_on('automake', type='build')
     depends_on('libtool',  type='build')
     depends_on('m4',       type='build')
 
-    # FIXME unnecessary dependency on python3 because of implicit python3
-    # dependency through meson by a dependency of wx/gtkplus
-    depends_on('python@3:')
-
+    depends_on('libharu')
     depends_on('wxwidgets')
-    depends_on('gdal')
-    depends_on('proj@:5')
-
+    depends_on('postgresql', when='+postgresql')
     depends_on('unixodbc', when='+odbc')
+
+    # SAGA-GIS requires projects.h from proj
+    depends_on('proj')
+    # https://sourceforge.net/p/saga-gis/bugs/271/
+    depends_on('proj@:5', when='@:7.2.999')
+
+    # Saga-Gis depends on legacy opencv API removed in opencv 4.x
+    depends_on('opencv@:3', when='+opencv')
+    # Set jpeg provider (similar to #8133)
+    depends_on('libjpeg', when='+opencv')
+    # Set hl variant due to similar issue #7145
+    depends_on('hdf5+hl')
```

Comment:
@tgamblin is currently working on a new concretizer. Hopefully it will resolve these issues.

Reply:
Is there a issue, which collects packages with this kind of hacky workaround? So that we know which packages to adjust as soon as the new concretizer is available?

Reply:
Nah, we'll just remove them as we go. No rush